WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Summary/SimplePaymentsSummaryViewModel.swift

```diff
@@ -56,6 +56,8 @@ final class SimplePaymentsSummaryViewModel: ObservableObject {
     ///
     let providedAmount: String
 
+    let amountName: String?
```

Comment:
This came up from the `test_update_simple_payments_order_sends_default_name_when_none_provided` test, as we set `amountName: nil` but we assert against `name: "Simple Payments"`:

By this property I would assume that a Simple Payments order either has a given string as name, or has no name, but without this property they still get the "Simple Payment" name by default, given in the `OrderFactory`. Should we rename this to some sort of "overrideFeeLineName", or similar? Perhaps move the default value somewhere else outside the factory? Perhaps make it non-nil? 🤔

Reply:
I don't think so... I don't feel super strongly, but I think it's best just to leave this as is.

At risk of bikeshedding a minor point in a part of the code which should be removed (because we don't do Simple Payments any more): the default is implemented in business logic. The UI/view model doesn't need to care about what happens in business logic in this case, just that the user has the option of providing an amount name, or not.

It makes sense to me to have a default in business logic – an unnamed fee would be confusing when reviewed.

Naming it `overrideFeeLineName` makes the view model harder to read/understand, and prompting questions that the callers probably don't need to care about or intuit – what's it overriding? Is it the `OverrideFee`'s line name, or is it an override for the `Fee`'s line name? Why's there a fee anyway, this is a payment? It feels like we're leaking implementation details with this sort of approach.

Making it non-nillable is simple, but would mean that we lose the ability to rely on a default value, and have to provide it separately for anything that uses that order action. 

Putting the default value elsewhere in code is probably the most compelling alternative, and I wouldn't be against doing that. It does mean that every caller of the relevant order action needs to look up or have its own default though.

Reply:
That makes sense, and I have no strong opinion about it neither, just cached my attention because of the test, and changing it feels too much hassle for such a little detail.

PD: Learned a new word today with `bikeshedding`, gonna use that one 😂

Reply:
https://en.wikipedia.org/wiki/Law_of_triviality